core/src/main/java/org/springframework/security/authentication/DelegatingAuthenticationManager.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authentication;
+
+import java.util.List;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationRegistry;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.CredentialsContainer;
+import org.springframework.util.Assert;
+
+public final class DelegatingAuthenticationManager implements AuthenticationManager {
+
+	private final List<AuthenticationManager> delegates;
+
+	private ObservationRegistry registry;
+
+	private AuthenticationEventPublisher eventPublisher = new NullEventPublisher();
+
+	private KeyValuesProvider keyValuesProvider = new KeyValuesProvider();
+
+	public DelegatingAuthenticationManager(List<AuthenticationManager> delegates) {
+		Assert.notEmpty(delegates, "delegates cannot be null or empty");
+		this.delegates = delegates;
+	}
+
+	@Override
+	public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+		ObservationContext context = new ObservationContext(authentication);
+		Observation observation = Observation.createNotStarted("spring.security.authentication", context, this.registry)
+				.keyValuesProvider(this.keyValuesProvider);
+		try (Observation.Scope scope = observation.openScope()) {
+			return authenticate(context, authentication);
+		}
+		catch (AuthenticationException ex) {
+			this.eventPublisher.publishAuthenticationFailure(ex, authentication);
+			observation.error(ex);
+			throw ex;
+		}
+		finally {
+			observation.stop();
+		}
```

Comment:
I'm not sure how sensitive the order is here, but you might be able to spare some boilerplate by doing something like this:

```suggestion
		try {
		    return observation.observeChecked(() -> authenticate(context, authentication));
		}
		catch (AuthenticationException ex) {
			this.eventPublisher.publishAuthenticationFailure(ex, authentication);
			throw ex;
		}
```

In this case you don't need to start the Observation so please ignore my comment above about start and `publishAuthenticationFailure` will be called after the observation was stopped (this also means that it is not measured). This seems fine to me though. 

Or you can move the error handling to the `authenticate` method.